linera-core/src/unit_tests/test_helpers.rs

```diff
@@ -1,57 +1,90 @@
+
+    match &**execution_error {
+        ExecutionError::SystemError(SystemExecutionError::InsufficientFunding { .. }) => {}
+        _ => panic!("Expected ExecutionError::SystemError::InsufficientFunding"),
+    }
 }
 
-#[cfg(test)]
+/// Asserts that an error is due to insufficient funding for fees.
 pub fn assert_insufficient_funding_fees<T>(obtained_error: &Result<T, ChainClientError>) {
-    assert_matches!(
-        obtained_error.as_ref().err().unwrap(),
-        ChainClientError::LocalNodeError(
-            LocalNodeError::WorkerError(WorkerError::ChainError(error))
-        ) if matches!(&**error, ChainError::ExecutionError(
-            execution_error, ChainExecutionContext::Block
-        ) if matches!(**execution_error, ExecutionError::SystemError(
-            SystemExecutionError::InsufficientFundingForFees { .. }
-        )))
-    );
+    let error = match obtained_error.as_ref().err().unwrap() {
+        ChainClientError::LocalNodeError(LocalNodeError::WorkerError(
+            WorkerError::ChainError(error),
+        )) => error,
+        _ => panic!("Expected a ChainClientError::LocalNodeError with a WorkerError::ChainError"),
+    };
+
+    let (execution_error, context) = match &**error {
+        ChainError::ExecutionError(execution_error, context) => (execution_error, context),
+        _ => panic!("Expected a ChainError::ExecutionError"),
+    };
+
+    if !matches!(context, ChainExecutionContext::Block) {
+        panic!("Expected ChainExecutionContext::Block, found {:?}", context);
+    }
+
+    match &**execution_error {
+        ExecutionError::SystemError(SystemExecutionError::InsufficientFundingForFees { .. }) => {}
+        _ => panic!("Expected ExecutionError::SystemError::InsufficientFundingForFees"),
+    }
 }
 
-#[cfg(test)]
+/// Asserts that an error is due to insufficient funding with a generic execution context.
 pub fn assert_insufficient_funding<T>(
     obtained_error: &Result<T, ChainClientError>,
     expected_context: ChainExecutionContext,
 ) {
-    assert_matches!(
-        obtained_error.as_ref().err().unwrap(),
+    let error = match obtained_error.as_ref().err().unwrap() {
         ChainClientError::LocalNodeError(LocalNodeError::WorkerError(
-            WorkerError::ChainError(error)
-        )) if matches!(&**error, ChainError::ExecutionError(
-            execution_error, ctx
-        ) if matches!(**execution_error, ExecutionError::SystemError(
-            SystemExecutionError::InsufficientFunding { .. }
-        )) && std::mem::discriminant(ctx) == std::mem::discriminant(&expected_context))
-    );
-}
+            WorkerError::ChainError(error),
+        )) => error,
+        _ => panic!("Expected a ChainClientError::LocalNodeError with a WorkerError::ChainError"),
+    };
+
+    let (execution_error, context) = match &**error {
+        ChainError::ExecutionError(execution_error, context) => (execution_error, context),
+        _ => panic!("Expected a ChainError::ExecutionError"),
+    };
```

Comment:
Idea: same as in the test above, using the `let .. else { .. };` pattern.

```suggestion
    let ChainError::ExecutionError(execution_error, context) = error {
        panic!("Expected a `ChainError::ExecutionError`, found: {error:#?}");
    };
```